### PR TITLE
Reproducible iteration order in DefaultAtlasCompoundClassLoader #2702

### LINKED ISSUE #2702: Reproducible iteration order in DefaultAtlasCompoundClassLoader
HashSet makes no guarantees as to the iteration order of the set. This matters if the class sets loadable by the individual delegate class loaders overlap. DefaultAtlasCompoundClassLoader thus may return different results on different JVM implementations or versions.

A PR follows.

### PR DESCRIPTION
**Pull request by ppalaga**
Fix #2702

### CODE REVIEW

**Comment by igarashitm** [comment]
Thanks for the PR, can you add `fix:` prefix to the commit message?
https://github.com/atlasmap/atlasmap/wiki/AtlasMap-Developer-Memo#commit-message

**Comment by ppalaga** [comment]
> can you add `fix:` prefix to the commit message?

Thanks, done

File: lib/core/src/main/java/io/atlasmap/core/DefaultAtlasCompoundClassLoader.java [hunk 1]
```diff
@@ -19,6 +19,7 @@
 import java.util.Collections;
 import java.util.Enumeration;
 import java.util.HashSet;
```
Comment (igarashitm) [inline lib/core/src/main/java/io/atlasmap/core/DefaultAtlasCompoundClassLoader.java:21]: It seems unused import checkstyle error is thrown for this one

Comment (ppalaga) [reply lib/core/src/main/java/io/atlasmap/core/DefaultAtlasCompoundClassLoader.java]: Thanks, fixed d36f617

**Review by igarashitm — APPROVED** [review]
:+1:

**Pull request merged**